src/main/java/de/tum/in/www1/artemis/repository/LectureRepository.java

```diff
@@ -100,6 +101,23 @@ public interface LectureRepository extends ArtemisJpaRepository<Lecture, Long> {
             """)
     Optional<Lecture> findByIdWithLectureUnitsAndSlidesAndAttachments(@Param("lectureId") long lectureId);
 
+    /**
+     * Finds a lecture by title and course id. Currently, name duplicates are allowed but this method throws an exception if multiple lectures with the
+     * same title are found.
+     *
+     * @param title    the title of the lecture
+     * @param courseId the id of the course containing the lecture
+     * @return the lecture with the given title and course id
+     * @throws NonUniqueResultException if multiple lectures with the same name in the same course are found
+     */
+    @Query("""
+            SELECT lecture
+            FROM Lecture lecture
+                LEFT JOIN FETCH lecture.lectureUnits
+            WHERE lecture.title = :title AND lecture.course.id = :courseId
+            """)
+    Optional<Lecture> findByTitleAndCourseIdWithLectureUnits(@Param("title") String title, @Param("courseId") long courseId) throws NonUniqueResultException;
```

Comment:
We should probably create an issue that specifies what course/lecture-level duplicates we want to prevent (and potentially migrate, as duplicates may already exist on production/test systems). Until that happens, these methods are potentially unsafe and, because the exception is unchecked, developers might not realize